frontend/src/test/scala/bloop/FileWatchingSpec.scala

```diff
@@ -188,6 +189,78 @@ object FileWatchingSpec extends BaseSuite {
     }
   }
 
+  flakyTest("support globs", 3) {
+    TestUtil.withinWorkspace { workspace =>
+      import ExecutionContext.ioScheduler
+      object Sources {
+        val `A.scala` =
+          """/A.scala
+            |class A
+          """.stripMargin
+
+        val `A2.scala` =
+          """/A.scala
+            |class A { val foo = 0 }
+          """.stripMargin
+      }
+
+      val baseProject = TestProject(workspace, "a", sources = List(Sources.`A.scala`))
+      val globDirectory = baseProject.config.directory.resolve("src")
+      val `A` = baseProject.copy(
+        config = baseProject.config.copy(
+          sources = Nil,
+          sourcesGlobs = Some(
+            List(
+              Config.SourcesGlobs(
+                globDirectory,
+                walkDepth = None,
+                includes = List("glob:*.scala"),
+                excludes = Nil
+              )
+            )
+          )
+        )
+      )
+      val projects = List(`A`)
+
+      val initialState = loadState(workspace, projects, new RecordingLogger())
+      val compiledState = initialState.compile(`A`)
+      assert(compiledState.status == ExitStatus.Ok)
+      assertValidCompilationState(compiledState, projects)
+
+      val (logObserver, logsObservable) =
+        Observable.multicast[(String, String)](MulticastStrategy.replay)(ioScheduler)
+      val logger = new PublisherLogger(logObserver, debug = true, DebugFilter.All)
+
+      val futureWatchedCompiledState =
+        compiledState.withLogger(logger).compileHandle(`A`, watch = true)
+
+      val HasIterationStoppedMsg = s"Watching ${numberDirsOf(compiledState.getDagFor(`A`))}"
+      def waitUntilIteration(totalIterations: Int, duration: Option[Long] = None): Task[Unit] =
+        waitUntilWatchIteration(logsObservable, totalIterations, HasIterationStoppedMsg, duration)
+
+      def testValidLatestState: TestState = {
+        val state = compiledState.getLatestSavedStateGlobally()
+        assert(state.status == ExitStatus.Ok)
+        assertValidCompilationState(state, projects)
+        state
+      }
+
+      TestUtil.await(FiniteDuration(20, TimeUnit.SECONDS), ExecutionContext.ioScheduler) {
```

Comment:
20 seconds, jeez 😅 this is a slow test!

Reply:
I copied everything over from the other, similar test in this file. On my machine it takes ~11s to complete 😞